ignite/contrib/handlers/tqdm_logger.py

```diff
@@ -0,0 +1,113 @@
+        Create a progress bar that shows you some metrics as they are computed,
+        by simply attaching the progress bar object to your engine.
+
+        .. code-block:: python
+
+            pbar = ProgressBar()
+            pbar.attach(trainer, len(data_loader), ['loss'])
+
+        If additionally, you want to keep track of any metric, add a call to ``attach_metrics``
+
+        .. code-block:: python
+
+            pbar.attach_metrics(trainer, ['loss', 'accuracy'], mode='epoch', log_interval=1)
+
+    Note:
+        When adding attaching the progress bar to an engine, it is recommend that you replace
+        every print operation in the engine's handlers with ``pbar.log_message`` to guarantee
+        the correct format of the stdout.
+    """
+
+    def __init__(self):
+        self.pbar = None
+
+    def _reset(self, engine, num_iterations):
+        self.pbar = tqdm(
+            total=num_iterations,
+            leave=False,
+            bar_format='{desc}[{n_fmt}/{total_fmt}] {percentage:3.0f}%|{bar}{postfix} [{elapsed}<{remaining}]')
+
+    def _close(self, engine):
+        self.pbar.close()
+
+    def _update(self, engine, metric_names):
+        metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+        self.pbar.set_description('Epoch {}'.format(engine.state.epoch))
+        self.pbar.set_postfix(**metrics)
+        self.pbar.update()
+
+    @staticmethod
+    def log_message(message):
+        """
+        Logs a message, preserving the progress bar correct output format
+
+        Args:
+            message (str): string you wish to log
+        """
+        tqdm.write(message)
+
+    def _log_metrics(self, engine, metric_names, mode='epoch', log_interval=1):
+
+        i = engine.state.epoch if mode == 'epoch' else engine.state.iteration
+
+        if log_interval and i % log_interval == 0:
+            if mode == 'epoch':
+                message = 'Epoch {}'.format(engine.state.epoch)
+            else:
+                message = 'Iteration {}'.format(engine.state.iteration)
+
+            metrics = {name: '{:.2e}'.format(engine.state.metrics[name]) for name in metric_names}
+            for name, value in metrics.items():
+                message += ' | {}={}'.format(name, value)
+
+            self.log_message(message)
+
+    def attach(self, engine, num_iterations, metric_names):
+        """
+        Attaches the progress bar to an engine object
+
+        Args:
+            engine (Engine): engine object
+            num_iterations (int): number of iterations of one epoch
+            metric_names (list): list of the metrics names to log
+        """
+
+        engine.add_event_handler(Events.EPOCH_STARTED, self._reset, num_iterations)
+        engine.add_event_handler(Events.EPOCH_COMPLETED, self._close)
+        engine.add_event_handler(Events.ITERATION_COMPLETED, self._update, metric_names)
+
+    def attach_metrics(self, engine, metric_names, mode='epoch', log_interval=1):
```

Comment:
IMO method name is unclear as `attach` and `atach_metrics` both requires `metric_names`...

Reply:
I wondered the same but didn't come up with a better one... Suggestions?

Reply:
> Suggestions?

`add_periodic_logging` or samething like that ?

Reply:
I thought of `log_metrics`, but didn't sound very solid to me

Reply:
changed it to `add_logging`, sounds good to me this way